docs/api/synthetics/monitors/update-monitor-api.asciidoc

```diff
@@ -0,0 +1,271 @@
+[[update-monitor-api]]
+== Update Monitor API
+++++
+<titleabbrev>Update monitor</titleabbrev>
+++++
+
+Updates a new monitor with the specified attributes. The required and default fields may vary based on the monitor type.
+
+=== {api-request-title}
+
+`PUT <kibana host>:<port>/api/synthetics/monitors/<config_id>`
+
+`PUT <kibana host>:<port>/s/<space_id>/api/synthetics/monitors/<config_id>`
+
+=== {api-prereq-title}
+
+You must have `all` privileges for the *Synthetics* feature in the *{observability}* section of the
+<<kibana-feature-privileges,{kib} feature privileges>>.
+
+
+[[update-monitor-api-path-params]]
+=== {api-path-parms-title}
+
+`config_id`::
+(Required, string) The ID of the monitor that you want to update.
+
+`space_id`::
+(Optional, string) An identifier for the space. If `space_id` is not provided in
+the URL, the default space is used.
+
+[[monitor-update-body]]
+==== Request Body
+
+The request body should contain the attributes of the monitor you want to update. The required and default fields differ depending on the monitor type:
+
+*Common Fields*:
+
+`name`:: (Required, string): The monitor's name.
+
+`type`:: (Required, string): The monitor type (e.g., `http` , `tcp` , `icmp` or `browser`) can't be changed after the monitor is created.
```

Comment:
Why is this listed here as a common field (and denoted as required?) when it can't be changed?

Including this field yields a `400` like:

```json
{"statusCode":400,"error":"Bad Request","message":"Monitor type cannot be changed from http to tcp.","attributes":{"details":"","type":"tcp","form_monitor_type":"http","enabled":true,"alert":{"status":{"enabled":true},"tls":{"enabled":true}},"schedule":3,"service.name":"test-service","config_id":"ddff2f1c-5cb3-43a5-9d94-3f4dbe59f081","tags":["prod"],"timeout":23,"name":"Test-http-monitor","locations":[{"id":"dev","label":"Dev Service","geo":{"lat":0,"lon":0},"isServiceManaged":true}],"namespace":"default","origin":"ui","journey_id":"","hash":"","id":"ddff2f1c-5cb3-43a5-9d94-3f4dbe59f081","params":"","max_attempts":2,"revision":2,"__ui":{"is_tls_enabled":false},"urls":"https://elastic.co","max_redirects":"0","url.port":null,"password":"","proxy_url":"","proxy_headers":{},"check.response.body.negative":[],"check.response.body.positive":[],"check.response.json":[],"response.include_body":"on_error","check.response.headers":{},"response.include_headers":true,"check.response.status":[],"check.request.body":{"value":"","type":"text"},"check.request.headers":{},"check.request.method":"GET","username":"","mode":"any","response.include_body_max_bytes":"1024","ipv4":true,"ipv6":true,"ssl.certificate_authorities":"","ssl.certificate":"","ssl.key":"","ssl.key_passphrase":"","ssl.verification_mode":"full","ssl.supported_protocols":["TLSv1.1","TLSv1.2","TLSv1.3"]}}
```

Reply:
https://github.com/elastic/kibana/pull/169928/commits/81c6928c9bca943e24a338481562609f2025e029